README.md

```diff
@@ -1,16 +1,305 @@
+processes that are costly, this can be a handy tool.
+
+## Define `set` Keyword in JavaScript Class Context
+
+In the previous section we saw we could write the following after employing `get`:
+
+```js
+let square = new Square(5);
+square.sideLength; // => 5
+square.area; // => 25
+```
+
+In the `Square` `class`, we only have one real property,
+`this.sideLength`. If we want to change that directly, we would write
+`this.sideLength =` whatever value we assign it.
+
+```js
+let square = new Square(5);
+square.sideLength; // => 5
+square.area; // => 25
+square.sideLength = 10;
+```
+
+Which will change the value returned by `area`:
+
+```js
+square.area; // => 100
+```
+
+At the moment, although `area` looks and behaves like a property, we can't
+assign it a new value like a real property. Using `get` to create a
+pseudo-prototype is only half the story, since it is only used for retrieving
+data from an instance.
+
+To change data, we have `set`.
+
+To make `area` fully act like a real property, we can create both `get` and
+`set` methods for it:
+
+```js
+class Square {
+	constructor(sideLength) {
+		this.sideLength = sideLength;
+	}
+
+	get area() {
+		return this.sideLength * this.sideLength;
+	}
+
+	set area(newArea) {
+		this.sideLength = Math.sqrt(newArea);
+	}
+}
+```
+
+The result is that we can now 'set' the pseudo-property, `area`, and modify
+`this.sideLength` based on a reverse of the calculation we used in `get`:
+
+```js
+let square = new Square(5);
+square.sideLength; // => 5
+square.area; // => 25
+square.area = 64;
+square.sideLength; // => 8
+square.area; // => 64
+```
+
+We can now interact with `area` as though it is a modifiable property, even
+though `area` is derived.
+
+Using `set`, we can set up _indirect_ control to real properties. From the outside,
+it looks like a property is being set, but behind the scenes, we can apply
+conditional statements:
+
+```js
+set area(newArea) {
+  if (newArea > 0) {
+    this.sideLength = Math.sqrt(newArea)
+  } else {
+    console.warn("Area cannot be less than 0");
+  }
+}
+```
+
+Using `set` works well when we want to treat properties as private:
+
+```js
+class Student {
+	constructor(firstName, lastName) {
+		this._firstName = this.sanitize(firstName);
+		this._lastName = this.sanitize(lastName);
+	}
+
+	get firstName() {
+		return this.capitalize(this._firstName);
+	}
+
+	set firstName(firstName) {
+		this._firstName = this.sanitize(firstName);
+	}
+
+	capitalize(string) {
+		// capitalizes first letter
+		return string.charAt(0).toUpperCase() + string.slice(1);
+	}
+
+	sanitize(string) {
+		// removes any non alpha-numeric characters except dash and single quotes (apostrophes)
+		return string.replace(/[^A-Za-z0-9-']+/g, '');
+	}
+}
+
+let student = new Student('Carr@ol-Ann', ')Freel*ing');
+student; // => Student { _firstName: 'Carrol-Ann', _lastName: 'Freeling' }
+
+student.firstName = 'Hea@)@(!$)ther';
+student.firstName; // => 'Heather'
+```
```

Comment:
Is there a way to use the same `class` example throughout? Switching from math to names might be jarring. Or start with names and then use the math example so it will lead more into the lab.